### PR TITLE
🔒️ Improve model permissions handling

### PR DESCRIPTION
**Pull request by MrKrisKrisu**

### CODE REVIEW

File: app/Http/Resources/UserResource.php [hunk 1]
```diff
@@ -26,7 +26,7 @@ public function toArray($request): array {
             'twitterUrl'        => $this->twitterUrl ?? null,
             'mastodonUrl'       => $this->mastodonUrl ?? null,
             'privateProfile'    => (bool) $this->private_profile,
-            'userInvisibleToMe' => (bool) $this->userInvisibleToMe,
+            'userInvisibleToMe' => (bool) \request()->user()->cannot('view', $this), //TODO: Is $this working here?
```
Comment (MrKrisKrisu) [inline app/Http/Resources/UserResource.php]: @HerrLevin Can you check this as you know how Resources work?

File: app/Http/Controllers/StatusController.php [hunk 2]
```diff
@@ -40,7 +40,8 @@ public static function getStatus(int $statusId): Status {
                                                        'trainCheckin.Destination',
                                                        'trainCheckin.HafasTrip',
                                                        'event')->withCount('likes')->firstOrFail();
-        if (!$status->user->userInvisibleToMe && (!$status->statusInvisibleToMe || $status->visibility == StatusVisibility::UNLISTED)) {
+
+        if (request()?->user()->can('view', $status)) {
```
Comment (HerrLevin) [inline app/Http/Controllers/StatusController.php]: unsure if this handles unlisted correctly.

File: app/Http/Controllers/TransportController.php [hunk 3]
```diff
@@ -390,10 +390,14 @@ public static function TrainCheckin($tripId,
 
         // check for other people on this train
         foreach ($trainCheckin->alsoOnThisConnection as $otherStatus) {
-            $otherStatus->user->notify(new UserJoinedConnection($status->id,
-                                                                $status->trainCheckin->HafasTrip->linename,
-                                                                $status->trainCheckin->Origin->name,
-                                                                $status->trainCheckin->Destination->name));
+            if ($otherStatus->user->can('view', $status)) {
+                $otherStatus->user->notify(new UserJoinedConnection(
```
Comment (HerrLevin) [inline app/Http/Controllers/TransportController.php]: This has to be reverted with #633, right?

File: app/Http/Resources/UserResource.php (hunk 1 above)
Comment (HerrLevin) [reply app/Http/Resources/UserResource.php]: i can only check this theoretically. atm it's not possible for me to check it inside my dev-environment.

**Review by HerrLevin — APPROVED** [review]

**Pull request merged**